Scripts/Runtime/Core/ScriptableObjectCollection.cs

```diff
@@ -229,6 +229,12 @@ public bool Contains(ScriptableObjectCollectionItem item)
             return items.Contains(item);
         }
 
+        public bool ContainsReferenceTo(ScriptableObjectCollectionItem item)
+        {
+            return items.Exists(collectionItem =>
+                collectionItem.TryGetReference(out ScriptableObjectReferenceItem reference) && reference.targetGuid == item.GUID);
```

Comment:
Use the property `TargetGuid` here

also would be great if we could use String.Equals(reference.TargetGuid, item.GUID, StringComparison.OrdinalIgnoreCulture)` here

Reply:
Sorry I did this quick refactor through rider but something clearly went wrong lol